py-polars/tests/unit/operations/test_cast.py

```diff
@@ -620,3 +620,11 @@ def test_cast_time_to_date() -> None:
     msg = "cannot cast `Time` to `Date`"
     with pytest.raises(pl.ComputeError, match=msg):
         s.cast(pl.Date)
+
+
+def test_cast_array_to_different_width() -> None:
+    s = pl.Series([[1, 2], [3, 4]], dtype=pl.Array(pl.Int8, 2))
+    with pytest.raises(
+        pl.InvalidOperationError, match="cannot cast Array to a different width"
+    ):
+        s.cast(pl.Array(pl.Int16, 3))
```

Comment:
I agree that this shouldn't be possible. But `4` should be. Is in this PR as well? And do we test that?

Reply:
This PR only prohibits all cast to a different width.

Do we really want `cast` to become a reshape operation which may change the number of rows? I don't think that's a good idea.

Reply:
Discussed with Ritchie and we do not want to go for the reshape functionality.